contentprovider.go

```diff
@@ -711,7 +712,89 @@ func sortChunkMatchesByScore(ms []ChunkMatch) {
 	sort.Sort(chunkMatchScoreSlice(ms))
 }
 
-// Sort a slice of results.
-func SortFilesByScore(ms []FileMatch) {
-	sort.Sort(fileMatchSlice(ms))
+// k = 60 is arbitrary but reportedly works well (RFF; Cormack et al., 2009)
+const k = 60
+
+// SortFiles sorts files matches. The order depends on the match score and, if
+// ranks are available, on the pre-computed offline ranks.
+//
+// Rankings derived from match scores and rank vectors are combined based on
+// "Reciprocal Rank Fusion" (RFF).
+func SortFiles(ms []FileMatch) {
+	sort.Sort(fileMatchesByScore(ms))
+
+	if hasRanks(ms) {
+		rffScore := make([]float64, len(ms))
+
+		for i := 0; i < len(ms); i++ {
+			rffScore[i] = 1 / (k + float64(i))
+		}
+
+		sort.Sort(fileMatchesByRank{fileMatches: ms, rffScore: rffScore})
+
+		for i := 0; i < len(ms); i++ {
+			rffScore[i] += 1 / (k + float64(i))
+		}
```

Comment:
minor: you iterate over ms, but only touch rffScore. I know they are the same size, but feels less surprising to do something like
```suggestion
		for i := range rffScore {
			rffScore[i] += 1 / (k + float64(i))
		}
```